cpp/arcticdb/version/version_core.cpp

```diff
@@ -479,9 +493,8 @@ SegmentInMemory read_direct(const std::shared_ptr<Store>& store,
     util::print_total_mem_usage(__FILE__, __LINE__, __FUNCTION__);
 
     ARCTICDB_DEBUG(log::version(), "Fetching frame data");
-    fetch_data(frame, pipeline_context, store, opt_false(read_options.dynamic_schema_), buffers).get();
-    util::print_total_mem_usage(__FILE__, __LINE__, __FUNCTION__);
-    return frame;
+    auto future_fetch_data = fetch_data(frame, pipeline_context, store, opt_false(read_options.dynamic_schema_), buffers).get();
```

Comment:
Wait, `get()` is still returning a future?

Reply:
amended this, was just the initial draft